llama.cpp

```diff
@@ -12727,6 +12727,10 @@ static std::vector<llama_grammar_candidate> llama_grammar_reject_candidates_for_
         }
     }
 
+    if (next_candidates.empty()) {
+        return rejects;
```

Comment:
Such an elegant optimization. Very nice catch!

Reply:
Took me forever (and many failed refactorings) to understand what the code was doing tbh 😅